### PR TITLE
PD-426: Upgrade Typescript to 3.7

### PR DESCRIPTION
**Pull request by brookescarlett**
## ✍️ Proposed changes

- Uprgrade Typescript and associated deps to 3.7
- Added babel plugins to support optional chaining and nullish coalescing 

🎟 _Jira ticket:_ [PD-426](https://jira.mongodb.org/browse/PD-426)

### CODE REVIEW

File: packages/icon-button/src/IconButton.tsx [hunk 1]
```diff
@@ -57,6 +57,12 @@ interface ButtonIconButtonProps
 
 type IconButtonProps = LinkIconButtonProps | ButtonIconButtonProps;
 
+function usesLinkElement(
```
Comment (hswolff) [inline packages/icon-button/src/IconButton.tsx]: What's with these changes?

Comment (brookescarlett) [reply packages/icon-button/src/IconButton.tsx]: Typescript started to complain about onClick handling. Have encountered this before with a component dynamically switching between 'a' and 'button' tags so just followed pattern that we've implemented in other places.

Comment (brookescarlett) [reply packages/icon-button/src/IconButton.tsx]: Should run a changeset for this component, great catch.

File: packages/icon-button/src/IconButton.tsx [hunk 2]
```diff
@@ -129,10 +135,12 @@ const iconButtonVariants: { readonly [K in Variant]: string } = {
 const disabledStyle: { readonly [K in Variant]: string } = {
   [Variant.Light]: css`
     color: ${uiColors.gray.light2};
+    pointer-events: none;
```
Comment (hswolff) [inline packages/icon-button/src/IconButton.tsx]: What about these changes?

Comment (brookescarlett) [reply packages/icon-button/src/IconButton.tsx]: Handles disabling pointer events when we're no longer conditionally implementing the onClick

**Pull request merged**